infrasim/model/elements/pcie_downstream.py

```diff
@@ -0,0 +1,50 @@
+'''
+*********************************************************
+Copyright @ 2015 EMC Corporation All Rights Reserved
+*********************************************************
+'''
+# -*- coding: utf-8 -*-
+
+
+from infrasim.model.core.element import CElement
+from infrasim import ArgsNotCorrect
+
+
+class CPCIEDownstream(CElement):
+    def __init__(self, downstream_info):
+        super(CPCIEDownstream, self).__init__()
+        self.__downstream_info = downstream_info
+        self.bus = None
+        self.id = None
+        self.__slot = None
+        self.addr = None
+        self.__device = None
+        self.__chassis = None
+        self.downstream_option = None
+
+    def set_option(self):
+        self.downstream_option = " -device {},id={},bus={},chassis={},slot={}".format(
+                                                            self.__device,
+                                                            self.id,
+                                                            self.bus,
+                                                            self.__chassis,
+                                                            self.__slot)
+
+    def precheck(self):
+        if self.__downstream_info is None:
+            self.logger.exception("[PCIEDownstream] \
+                Downstream device is required.")
+            raise ArgsNotCorrect("downstream device is required.")
+        if not set(['id', 'bus', 'chassis', 'slot']).issubset(self.__downstream_info):
+            self.logger.exception("[PCIEDownstream] \
```

Comment:
Don't have to log now. Every InfrasimError, including ArgsNotCorrect, will take one log entry when they are initialized, see InfrasimError for detail.